### ISSUE TITLE
Regression in pre update hook

### ISSUE THREAD

**Issue by clozanosanchez** [Automattic/mongoose#5141]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
In version 4.9.2 & 4.9.3, my pre update hooks are not being executed anymore. They run perfectly in version 4.9.1, if I install it again.

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Add a pre update hook to a schema:
```
schema.pre('update', myPreHook);
       
function myPreHook(next){
      console.log('PRE UPDATED')
      const query = this;
      ...
      next();
}
```
2. Execute an update operation:
```
const model = mongoose.model('test', schema);

model.update(updateQuery, {$set: updateData}, {runValidators: true}).exec()
   .then((updateResult) => {
        console.log('UPDATED')
        ...
    }).catch((error) => {
        ...
    });
```
3. The console writes 'UPDATED', but never 'PRE UPDATED'.

**What is the expected behavior?**
Pre update hook should be executed before executing the update operation.

**Please mention your node.js, mongoose and MongoDB version.**
Node.js: 6.10.1
Mongoose: 4.9.2 & 4.9.3
MongoDB: 2.2.25

**Comment by varunjayaraman** [comment]
do you define your hooks before calling `mongoose.model()`?

**Comment by clozanosanchez** [comment]
Yes, we define our hooks before calling mongoose.model()

**Comment by clozanosanchez** [comment]
We think it is related to #5098 (that was fixed in version 4.9.2). 
We are cloning the schema and then we execute the model method. Maybe the prehooks are not being cloned properly.

```
....

initialSchema.pre('update', myPreHook);
       
function myPreHook(next){
      console.log('PRE UPDATED')
      const query = this;
      ...
      next();
}

....

const clonedSchema = initialSchema.clone();
const model = mongoose.model('test', clonedSchema);

model.update(updateQuery, {$set: updateData}, {runValidators: true}).exec()
   .then((updateResult) => {
        console.log('UPDATED')
        ...
    }).catch((error) => {
        ...
    });
```
